sympy/physics/mechanics/jointsmethod.py

```diff
@@ -0,0 +1,222 @@
+from sympy.physics.mechanics import Body, Lagrangian, KanesMethod, LagrangesMethod
+from sympy.physics.mechanics.method import _Methods
+
+__all__ = ['JointsMethod']
+
+
+class JointsMethod(_Methods):
+    """Method for formulating the equations of motion using a set of interconnected bodies with joints.
+
+    Parameters
+    ==========
+
+    body : Body or ReferenceFrame
+        The inertial(Newtonion) frame.
+    *joints : Joint
+        The joints in the system
+
+    Attributes
+    ==========
+
+    q, u : Matrix
+        Matrices of the generalized coordinates and speeds
+    bodies : iterable
+        Iterable of Body objects in the system.
+    forcelist : iterable
+        Iterable of (Point, vector) or (ReferenceFrame, vector) tuples
+        describing the forces on the system.
+    mass_matrix : Matrix
+        The system's mass matrix
+    forcing : Matrix
+        The system's forcing vector
+    mass_matrix_full : Matrix
+        The "mass matrix" for the u's and q's
+    forcing_full : Matrix
+        The "forcing vector" for the u's and q's
+    method : KanesMethod or Lagrange's method
+        Method's object.
+
+    Examples
+    ========
+
+    This is a simple example for a one degree of freedom translational
+    spring-mass-damper.
+
+    >>> from sympy import symbols
+    >>> from sympy.physics.mechanics import Body, JointsMethod, PrismaticJoint
+    >>> from sympy.physics.vector import dynamicsymbols
+    >>> c, k = symbols('c k')
+    >>> q, u = dynamicsymbols('q u')
+    >>> W = Body('W')
```

Comment:
```suggestion
    >>> wall = Body('W')
```

more descriptive for the reader

Reply:
Wouldn't be adding `#Wall` as comment better?

Reply:
I try to use informative variable names so you don't need comments. The code can then speak for itself.

Reply:
Here's a short note on the topic: https://wiki.c2.com/?GoodVariableNames you can google more recommendations about this.